olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -153,112 +150,116 @@ bool DiskCache::Clear() {
 
-  if (!status.ok() && !isReadOnly)
-    OLP_SDK_LOG_WARNING(kLogTag, "Cannot open database ("
-                                     << status.ToString()
-                                     << ") attempting repair");
+  if (!status.ok() && !is_read_only)
+    OLP_SDK_LOG_WARNING(kLogTag, "Open: failed, attempting repair, error="
+                                     << status.ToString());
 
-  // if the database is r/w and corrupted, attempt to repair & reopen
-  if ((status.IsCorruption() || status.IsIOError()) && !isReadOnly &&
-      repairCache(versionedDataPath) == true) {
-    status = leveldb::DB::Open(options, versionedDataPath, &db);
+  // If the database is r/w and corrupted, attempt to repair & reopen
+  if ((status.IsCorruption() || status.IsIOError()) && !is_read_only &&
+      RepairCache(versioned_data_path) == true) {
+    status = leveldb::DB::Open(open_options, versioned_data_path, &db);
     if (status.ok()) {
       database_.reset(db);
       return OpenResult::Repaired;
     }
   }
 
   if (!status.ok()) {
-    setOpenError(status);
+    SetOpenError(status);
     return OpenResult::Fail;
   }
 
   database_.reset(db);
   return OpenResult::Success;
 }
 
-void DiskCache::setOpenError(const leveldb::Status& status) {
-  ErrorCode code = ErrorCode::Unknown;
-  if (status.IsNotFound()) code = ErrorCode::NotFound;
-  if (status.IsInvalidArgument()) code = ErrorCode::InvalidArgument;
-  if (status.IsCorruption() || status.IsIOError())
-    code = ErrorCode::InternalFailure;
-  if (status.IsNotSupportedError()) code = ErrorCode::BadRequest;
-  std::string errorMessage = status.ToString();
-  OLP_SDK_LOG_FATAL(kLogTag, "Cannot open database " << errorMessage);
-  error_ = ApiError(code, std::move(errorMessage));
+void DiskCache::SetOpenError(const leveldb::Status& status) {
+  client::ErrorCode code = client::ErrorCode::Unknown;
+  if (status.IsNotFound()) {
+    code = client::ErrorCode::NotFound;
+  }
+  if (status.IsInvalidArgument()) {
+    code = client::ErrorCode::InvalidArgument;
+  }
+  if (status.IsCorruption() || status.IsIOError()) {
+    code = client::ErrorCode::InternalFailure;
+  }
+  if (status.IsNotSupportedError()) {
+    code = client::ErrorCode::BadRequest;
+  }
+  std::string error_message = status.ToString();
+  OLP_SDK_LOG_ERROR(kLogTag, "Open: failed, error=" << error_message);
+  error_ = client::ApiError(code, std::move(error_message));
 }
 
 bool DiskCache::Put(const std::string& key, const std::string& value) {
-  if (!database_) return false;
+  if (!database_) {
+    return false;
+  }
 
-  if (environment_ && (max_size_ != std::uint64_t(-1))) {
-    if (environment_->Size() >= max_size_) return false;
+  if (environment_ && (max_size_ != kSizeMax) &&
+      (environment_->Size() >= max_size_)) {
+    return false;
   }
 
-  const leveldb::Status& status = database_->Put(
-      leveldb::WriteOptions(), toLeveldbSlice(key), toLeveldbSlice(value));
+  const auto status = database_->Put(
```

Comment:
```suggestion
  const auto& status = database_->Put(
```

Reply:
Not sure why it was like that as the leveldb::Put() method has the following signature:

```cpp
Status DB::Put(const WriteOptions& opt, const Slice& key, const Slice& value)
```

So theoretically we used a ref to a temp variable before if the compiler did not optimized it out.

Reply:
Ok. Either should be fine.